In [1]:
import pandas as pd

In [ ]:
# rf = pd.DataFrame()
# for i in range(1,46,1):
#     tune = pd.read_csv('../results/tuning/tuning_pieces/rf%d_tuning.csv' %i)
#     tune.drop(columns=['Unnamed: 0', 'max_features'], inplace=True)
#     rf = pd.concat([rf, tune]).reset_index(drop=True)
# rf.to_csv('../results/tuning/rf_tuning.csv')

In [2]:
rf = pd.read_csv('../results/tuning/rf_tuning.csv')
rf.drop(columns='Unnamed: 0', inplace=True)

rf_count = rf.loc[(rf['vector'] == 'count') & (rf['dataset'] == 'val')].copy()
rf_count.drop(columns='fold', inplace=True)
rf_count = rf_count.groupby(['ngram_range','max_df','min_df','n_estimators','criterion','max_depth']).mean()
rf_count_acc = rf_count.sort_values('Acc', ascending=False).head(1).reset_index()
rf_count_auc = rf_count.sort_values('AUC', ascending=False).head(1).reset_index()
print('Random Forest - Count Vectorizer, Best parameters for accuracy:')
print('ngram_range=%s' %str(rf_count_acc['ngram_range'].values[0]))
print('max_df=%f' %rf_count_acc['max_df'].values[0])
print('min_df=%f' %rf_count_acc['min_df'].values[0])
print('n_estimators=%d' %rf_count_acc['n_estimators'].values[0])
print('criterion=%s' %rf_count_acc['criterion'].values[0])
print('max_depth=%d' %rf_count_acc['max_depth'].values[0])
print('Scores: Acc=%.04f, AUC=%.04f, Recall=%.04f, PPV=%.04f' %(rf_count_acc['Acc'].values[0], 
                                                                rf_count_acc['AUC'].values[0], 
                                                                rf_count_acc['recall'].values[0], 
                                                                rf_count_acc['PPV'].values[0]))
print('')

print('Random Forest - Count Vectorizer, Best parameters for AUC:')
print('ngram_range=%s' %str(rf_count_auc['ngram_range'].values[0]))
print('max_df=%f' %rf_count_auc['max_df'].values[0])
print('min_df=%f' %rf_count_auc['min_df'].values[0])
print('n_estimators=%d' %rf_count_auc['n_estimators'].values[0])
print('criterion=%s' %rf_count_auc['criterion'].values[0])
print('max_depth=%d' %rf_count_auc['max_depth'].values[0])
print('Scores: Acc=%.04f, AUC=%.04f, Recall=%.04f, PPV=%.04f' %(rf_count_auc['Acc'].values[0], 
                                                                rf_count_auc['AUC'].values[0], 
                                                                rf_count_auc['recall'].values[0], 
                                                                rf_count_auc['PPV'].values[0]))
print('')
print('')


rf_tfidf = rf.loc[(rf['vector'] == 'tfidf') & (rf['dataset'] == 'val')].copy()
rf_tfidf.drop(columns='fold', inplace=True)
rf_tfidf = rf_tfidf.groupby(['ngram_range','max_df','min_df','n_estimators','criterion','max_depth']).mean()
rf_tfidf_acc = rf_tfidf.sort_values('Acc', ascending=False).head(1).reset_index()
rf_tfidf_auc = rf_tfidf.sort_values('AUC', ascending=False).head(1).reset_index()
print('Random Forest - TF-IDF Vectorizer, Best parameters for accuracy:')
print('ngram_range=%s' %str(rf_tfidf_acc['ngram_range'].values[0]))
print('max_df=%f' %rf_tfidf_acc['max_df'].values[0])
print('min_df=%f' %rf_tfidf_acc['min_df'].values[0])
print('n_estimators=%d' %rf_tfidf_acc['n_estimators'].values[0])
print('criterion=%s' %rf_tfidf_acc['criterion'].values[0])
print('max_depth=%d' %rf_tfidf_acc['max_depth'].values[0])
print('Scores: Acc=%.04f, AUC=%.04f, Recall=%.04f, PPV=%.04f' %(rf_tfidf_acc['Acc'].values[0], 
                                                                rf_tfidf_acc['AUC'].values[0], 
                                                                rf_tfidf_acc['recall'].values[0], 
                                                                rf_tfidf_acc['PPV'].values[0]))
print('')

print('Random Forest - TF-IDF Vectorizer, Best parameters for AUC:')
print('ngram_range=%s' %str(rf_tfidf_auc['ngram_range'].values[0]))
print('max_df=%f' %rf_tfidf_auc['max_df'].values[0])
print('min_df=%f' %rf_tfidf_auc['min_df'].values[0])
print('n_estimators=%d' %rf_tfidf_auc['n_estimators'].values[0])
print('criterion=%s' %rf_tfidf_auc['criterion'].values[0])
print('max_depth=%d' %rf_tfidf_auc['max_depth'].values[0])
print('Scores: Acc=%.04f, AUC=%.04f, Recall=%.04f, PPV=%.04f' %(rf_tfidf_auc['Acc'].values[0], 
                                                                rf_tfidf_auc['AUC'].values[0], 
                                                                rf_tfidf_auc['recall'].values[0], 
                                                                rf_tfidf_auc['PPV'].values[0]))

Random Forest - Count Vectorizer, Best parameters for accuracy:
ngram_range=(1, 1)
max_df=1.000000
min_df=0.000000
n_estimators=300
criterion=gini
max_depth=100
Scores: Acc=0.8601, AUC=0.9389, Recall=0.9313, PPV=0.8209

Random Forest - Count Vectorizer, Best parameters for AUC:
ngram_range=(1, 1)
max_df=0.750000
min_df=0.000000
n_estimators=300
criterion=entropy
max_depth=100
Scores: Acc=0.8597, AUC=0.9407, Recall=0.9353, PPV=0.8182


Random Forest - TF-IDF Vectorizer, Best parameters for accuracy:
ngram_range=(1, 1)
max_df=0.500000
min_df=0.000000
n_estimators=300
criterion=entropy
max_depth=100
Scores: Acc=0.8601, AUC=0.9403, Recall=0.9325, PPV=0.8203

Random Forest - TF-IDF Vectorizer, Best parameters for AUC:
ngram_range=(1, 1)
max_df=1.000000
min_df=0.000000
n_estimators=300
criterion=entropy
max_depth=100
Scores: Acc=0.8596, AUC=0.9405, Recall=0.9317, PPV=0.8201


### Random Forest:
For Count Vectorizer:
<ul>
    <li>ngram_range: (1, 1)</li>
    <li>max_df: 1.0</li>
    <li>min_df: 0</li>
    <li>n_estimators: 300</li>
    <li>criterion: gini</li>
    <li>max_depth: 100</li>
</ul>
<br/>
<br/>
For TF-IDF Vectorizer:
<ul>
    <li>ngram_range: (1, 1)</li>
    <li>max_df: 0.5</li>
    <li>min_df: 0</li>
    <li>n_estimators: 300</li>
    <li>criterion: entropy</li>
    <li>max_depth: 100</li>
</ul>

In [4]:
# lsvm = pd.DataFrame()
# for i in range(1,10,1):
#     tune = pd.read_csv('../results/tuning/tuning_pieces/tuning_svm_count/temp_lsvm%d_tuning.csv' %i)
#     tune.drop(columns=['Unnamed: 0', 'max_features'], inplace=True)
#     tune['kernel'] = 'linear'
#     lsvm = pd.concat([lsvm, tune]).reset_index(drop=True)
# for i in range(1,4,1):
#     tune = pd.read_csv('../results/tuning/tuning_pieces/tuning_svm_tfidf/temp_lsvm%d_tuning.csv' %i)
#     tune.drop(columns=['Unnamed: 0'], inplace=True)
#     tune['kernel'] = 'linear'
#     lsvm = pd.concat([lsvm, tune]).reset_index(drop=True)
    
# rsvm = pd.DataFrame()
# for i in range(1,10,1):
#     tune = pd.read_csv('../results/tuning/tuning_pieces/tuning_svm_count/temp_rsvm%d_tuning.csv' %i)
#     tune.drop(columns=['Unnamed: 0', 'max_features'], inplace=True)
#     tune['kernel'] = 'rbf'
#     rsvm = pd.concat([rsvm, tune]).reset_index(drop=True)
# for i in range(1,4,1):
#     tune = pd.read_csv('../results/tuning/tuning_pieces/tuning_svm_tfidf/temp_rsvm%d_tuning.csv' %i)
#     tune.drop(columns=['Unnamed: 0'], inplace=True)
#     tune['kernel'] = 'rbf'
#     rsvm = pd.concat([rsvm, tune]).reset_index(drop=True)
    
# psvm = pd.DataFrame()
# for i in range(1,10,1):
#     tune = pd.read_csv('../results/tuning/tuning_pieces/tuning_svm_count/temp_psvm%d_tuning.csv' %i)
#     tune.drop(columns=['Unnamed: 0', 'max_features'], inplace=True)
#     tune['kernel'] = 'poly'
#     psvm = pd.concat([psvm, tune]).reset_index(drop=True)
# for i in range(1,4,1):
#     tune = pd.read_csv('../results/tuning/tuning_pieces/tuning_svm_tfidf/temp_psvm%d_tuning.csv' %i)
#     tune.drop(columns=['Unnamed: 0'], inplace=True)
#     tune['kernel'] = 'poly'
#     psvm = pd.concat([psvm, tune]).reset_index(drop=True)
        
# svm = pd.concat([lsvm, rsvm, psvm]).reset_index(drop=True)
# svm.to_csv('../results/tuning/svm_tuning.csv')

In [7]:
svm = pd.read_csv('../results/tuning/svm_tuning.csv')
svm.drop(columns='Unnamed: 0', inplace=True)

svm_count = svm.loc[(svm['vector'] == 'count') & (svm['dataset'] == 'val')].copy()
svm_count.drop(columns='fold', inplace=True)
svm_count = svm_count.groupby(['ngram_range','max_df','min_df','C','kernel']).mean()
svm_count_acc = svm_count.sort_values('Acc', ascending=False).head(1).reset_index()
svm_count_auc = svm_count.sort_values('AUC', ascending=False).head(1).reset_index()
print('SVM - Count Vectorizer, Best parameters for accuracy:')
print('ngram_range=%s' %str(svm_count_acc['ngram_range'].values[0]))
print('max_df=%f' %svm_count_acc['max_df'].values[0])
print('min_df=%f' %svm_count_acc['min_df'].values[0])
print('C=%f' %svm_count_acc['C'].values[0])
print('kernel=%s' %svm_count_acc['kernel'].values[0])
print('Scores: Acc=%.04f, AUC=%.04f, Recall=%.04f, PPV=%.04f' %(svm_count_acc['Acc'].values[0], 
                                                                svm_count_acc['AUC'].values[0], 
                                                                svm_count_acc['recall'].values[0], 
                                                                svm_count_acc['PPV'].values[0]))
print('')

print('SVM - Count Vectorizer, Best parameters for AUC:')
print('ngram_range=%s' %str(svm_count_auc['ngram_range'].values[0]))
print('max_df=%f' %svm_count_auc['max_df'].values[0])
print('min_df=%f' %svm_count_auc['min_df'].values[0])
print('C=%f' %svm_count_auc['C'].values[0])
print('kernel=%s' %svm_count_auc['kernel'].values[0])
print('Scores: Acc=%.04f, AUC=%.04f, Recall=%.04f, PPV=%.04f' %(svm_count_auc['Acc'].values[0], 
                                                                svm_count_auc['AUC'].values[0], 
                                                                svm_count_auc['recall'].values[0], 
                                                                svm_count_auc['PPV'].values[0]))
print('')
print('')


svm_tfidf = svm.loc[(svm['vector'] == 'tfidf') & (svm['dataset'] == 'val')].copy()
svm_tfidf.drop(columns='fold', inplace=True)
svm_tfidf = svm_tfidf.groupby(['ngram_range','max_df','min_df','C','kernel']).mean()
svm_tfidf_acc = svm_tfidf.sort_values('Acc', ascending=False).head(1).reset_index()
svm_tfidf_auc = svm_tfidf.sort_values('AUC', ascending=False).head(1).reset_index()
print('SVM - TF-IDF Vectorizer, Best parameters for accuracy:')
print('ngram_range=%s' %str(svm_tfidf_acc['ngram_range'].values[0]))
print('max_df=%f' %svm_tfidf_acc['max_df'].values[0])
print('min_df=%f' %svm_tfidf_acc['min_df'].values[0])
print('C=%f' %svm_tfidf_acc['C'].values[0])
print('kernel=%s' %svm_tfidf_acc['kernel'].values[0])
print('Scores: Acc=%.04f, AUC=%.04f, Recall=%.04f, PPV=%.04f' %(svm_tfidf_acc['Acc'].values[0], 
                                                                svm_tfidf_acc['AUC'].values[0], 
                                                                svm_tfidf_acc['recall'].values[0], 
                                                                svm_tfidf_acc['PPV'].values[0]))
print('')

print('SVM - TF-IDF Vectorizer, Best parameters for AUC:')
print('ngram_range=%s' %str(svm_tfidf_auc['ngram_range'].values[0]))
print('max_df=%f' %svm_tfidf_auc['max_df'].values[0])
print('min_df=%f' %svm_tfidf_auc['min_df'].values[0])
print('C=%f' %svm_tfidf_auc['C'].values[0])
print('kernel=%s' %svm_tfidf_auc['kernel'].values[0])
print('Scores: Acc=%.04f, AUC=%.04f, Recall=%.04f, PPV=%.04f' %(svm_tfidf_auc['Acc'].values[0], 
                                                                svm_tfidf_auc['AUC'].values[0], 
                                                                svm_tfidf_auc['recall'].values[0], 
                                                                svm_tfidf_auc['PPV'].values[0]))

SVM - Count Vectorizer, Best parameters for accuracy:
ngram_range=(1, 2)
max_df=0.500000
min_df=0.000000
C=0.010000
kernel=linear
Scores: Acc=0.8476, AUC=0.9392, Recall=0.9410, PPV=0.7988

SVM - Count Vectorizer, Best parameters for AUC:
ngram_range=(1, 2)
max_df=0.500000
min_df=0.000000
C=0.010000
kernel=linear
Scores: Acc=0.8476, AUC=0.9392, Recall=0.9410, PPV=0.7988


SVM - TF-IDF Vectorizer, Best parameters for accuracy:
ngram_range=(1, 2)
max_df=0.250000
min_df=0.000000
C=1.257500
kernel=linear
Scores: Acc=0.8720, AUC=0.9454, Recall=0.8962, PPV=0.8609

SVM - TF-IDF Vectorizer, Best parameters for AUC:
ngram_range=(1, 2)
max_df=0.250000
min_df=0.000000
C=1.257500
kernel=linear
Scores: Acc=0.8720, AUC=0.9454, Recall=0.8962, PPV=0.8609


### SVM:
For Count Vectorizer:
<ul>
    <li>ngram_range: (1, 2)</li>
    <li>max_df: 0.5</li>
    <li>min_df: 0</li>
    <li>kernel: linear</li>
    <li>C: 0.01</li>
</ul>
<br/>
<br/>
For TF-IDF Vectorizer:
<ul>
    <li>ngram_range: (1, 2)</li>
    <li>max_df: 0.25</li>
    <li>min_df: 0</li>
    <li>kernel: linear</li>
    <li>C: 1.2575</li>
</ul>

In [2]:
# nn = pd.DataFrame()
# count = 0
# for i in range(1,46,1):
#     try:
#         tune = pd.read_csv('../results/tuning/tuning_pieces/nn%d_tuning.csv' %i)
#     except:
#         continue
#     else:
#         count += 1
#         tune.drop(columns=['Unnamed: 0'], inplace=True)
#         nn = pd.concat([nn, tune]).reset_index(drop=True)
        
# for i in range(1,46,1):
#     try:
#         tune = pd.read_csv('../results/tuning/tuning_pieces/temp_nn%d_tuning.csv' %i)
#     except:
#         continue
#     else:
#         count += 1
#         tune.drop(columns=['Unnamed: 0'], inplace=True)
#         nn = pd.concat([nn, tune]).reset_index(drop=True)

# for i in range(1,46,1):
#     try:
#         tune = pd.read_csv('../results/tuning/tuning_pieces/temp_nn%d_tuning_1.csv' %i)
#     except:
#         continue
#     else:
#         count += 1
#         tune.drop(columns=['Unnamed: 0'], inplace=True)
#         nn = pd.concat([nn, tune]).reset_index(drop=True)

# # extra pieces        
# tune = pd.read_csv('../results/tuning/tuning_pieces/temp_nn36_tuning_2.csv')
# count += 1
# tune.drop(columns=['Unnamed: 0'], inplace=True)
# nn = pd.concat([nn, tune]).reset_index(drop=True)

# nn.to_csv('../results/tuning/nn_tuning.csv')

In [4]:
nn = pd.read_csv('../results/tuning/nn_tuning.csv')
nn.drop(columns='Unnamed: 0', inplace=True)

nn_count = nn.loc[(nn['vector'] == 'count') & (nn['dataset'] == 'val')].copy()
nn_count.drop(columns='fold', inplace=True)
nn_count = nn_count.groupby(['ngram_range','max_df','min_df','h1_nodes','optimizer']).mean()
nn_count_acc = nn_count.sort_values('Acc', ascending=False).head(1).reset_index()
nn_count_auc = nn_count.sort_values('AUC', ascending=False).head(1).reset_index()
print('NN - Count Vectorizer, Best parameters for accuracy:')
print('ngram_range=%s' %str(nn_count_acc['ngram_range'].values[0]))
print('max_df=%f' %nn_count_acc['max_df'].values[0])
print('min_df=%f' %nn_count_acc['min_df'].values[0])
print('h1_nodes=%f' %nn_count_acc['h1_nodes'].values[0])
print('optimizer=%s' %nn_count_acc['optimizer'].values[0])
print('Scores: Acc=%.04f, AUC=%.04f, Recall=%.04f, PPV=%.04f' %(nn_count_acc['Acc'].values[0], 
                                                                nn_count_acc['AUC'].values[0], 
                                                                nn_count_acc['recall'].values[0], 
                                                                nn_count_acc['PPV'].values[0]))
print('')

print('NN - Count Vectorizer, Best parameters for AUC:')
print('ngram_range=%s' %str(nn_count_auc['ngram_range'].values[0]))
print('max_df=%f' %nn_count_auc['max_df'].values[0])
print('min_df=%f' %nn_count_auc['min_df'].values[0])
print('h1_nodes=%f' %nn_count_auc['h1_nodes'].values[0])
print('optimizer=%s' %nn_count_auc['optimizer'].values[0])
print('Scores: Acc=%.04f, AUC=%.04f, Recall=%.04f, PPV=%.04f' %(nn_count_auc['Acc'].values[0], 
                                                                nn_count_auc['AUC'].values[0], 
                                                                nn_count_auc['recall'].values[0], 
                                                                nn_count_auc['PPV'].values[0]))
print('')
print('')


nn_tfidf = nn.loc[(nn['vector'] == 'tfidf') & (nn['dataset'] == 'val')].copy()
nn_tfidf.drop(columns='fold', inplace=True)
nn_tfidf = nn_tfidf.groupby(['ngram_range','max_df','min_df','h1_nodes','optimizer']).mean()
nn_tfidf_acc = nn_tfidf.sort_values('Acc', ascending=False).head(1).reset_index()
nn_tfidf_auc = nn_tfidf.sort_values('AUC', ascending=False).head(1).reset_index()
print('NN - TF-IDF Vectorizer, Best parameters for accuracy:')
print('ngram_range=%s' %str(nn_tfidf_acc['ngram_range'].values[0]))
print('max_df=%f' %nn_tfidf_acc['max_df'].values[0])
print('min_df=%f' %nn_tfidf_acc['min_df'].values[0])
print('h1_nodes=%f' %nn_tfidf_acc['h1_nodes'].values[0])
print('optimizer=%s' %nn_tfidf_acc['optimizer'].values[0])
print('Scores: Acc=%.04f, AUC=%.04f, Recall=%.04f, PPV=%.04f' %(nn_tfidf_acc['Acc'].values[0], 
                                                                nn_tfidf_acc['AUC'].values[0], 
                                                                nn_tfidf_acc['recall'].values[0], 
                                                                nn_tfidf_acc['PPV'].values[0]))
print('')

print('NN - TF-IDF Vectorizer, Best parameters for AUC:')
print('ngram_range=%s' %str(nn_tfidf_auc['ngram_range'].values[0]))
print('max_df=%f' %nn_tfidf_auc['max_df'].values[0])
print('min_df=%f' %nn_tfidf_auc['min_df'].values[0])
print('h1_nodes=%f' %nn_tfidf_auc['h1_nodes'].values[0])
print('optimizer=%s' %nn_tfidf_auc['optimizer'].values[0])
print('Scores: Acc=%.04f, AUC=%.04f, Recall=%.04f, PPV=%.04f' %(nn_tfidf_auc['Acc'].values[0], 
                                                                nn_tfidf_auc['AUC'].values[0], 
                                                                nn_tfidf_auc['recall'].values[0], 
                                                                nn_tfidf_auc['PPV'].values[0]))

NN - Count Vectorizer, Best parameters for accuracy:
ngram_range=(1, 2)
max_df=1.000000
min_df=0.000000
h1_nodes=128.000000
optimizer=Adadelta
Scores: Acc=0.8678, AUC=0.9440, Recall=0.8920, PPV=0.8570

NN - Count Vectorizer, Best parameters for AUC:
ngram_range=(1, 1)
max_df=0.500000
min_df=0.000000
h1_nodes=3200.000000
optimizer=Adadelta
Scores: Acc=0.8651, AUC=0.9448, Recall=0.8965, PPV=0.8502


NN - TF-IDF Vectorizer, Best parameters for accuracy:
ngram_range=(1, 2)
max_df=0.750000
min_df=0.000000
h1_nodes=128.000000
optimizer=Adadelta
Scores: Acc=0.8706, AUC=0.9475, Recall=0.8889, PPV=0.8643

NN - TF-IDF Vectorizer, Best parameters for AUC:
ngram_range=(1, 2)
max_df=1.000000
min_df=0.000000
h1_nodes=128.000000
optimizer=Adadelta
Scores: Acc=0.8671, AUC=0.9481, Recall=0.8602, PPV=0.8793


### NN:
For Count Vectorizer:
<ul>
    <li>ngram_range: (1, 2)</li>
    <li>max_df: 1.0</li>
    <li>min_df: 0</li>
    <li>h1_nodes: 128</li>
    <li>optimizer: Adadelta</li>
</ul>
<br/>
<br/>
For TF-IDF Vectorizer:
<ul>
    <li>ngram_range: (1, 2)</li>
    <li>max_df: 0.75</li>
    <li>min_df: 0</li>
    <li>h1_nodes: 128</li>
    <li>optimizer: Adadelta</li>
</ul>